gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.ts

```diff
@@ -0,0 +1,317 @@
+import { MatInput } from '@angular/material/input';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { PortalSettings } from '../../../../entities/portal/portalSettings';
+import { ApiPortalHeader } from '../../../../entities/apiPortalHeader';
+import { EnvironmentApiHeadersService } from '../../../../services-ngx/environment-api-headers.service';
+import { SnackBarService } from '../../../../services-ngx/snack-bar.service';
+import { PortalSettingsService } from '../../../../services-ngx/portal-settings.service';
+import { GioPermissionService } from '../../../../shared/components/gio-permission/gio-permission.service';
+import {
+  ApiPortalHeaderDialogData,
+  ApiPortalHeaderEditDialogComponent,
+  ApiPortalHeaderEditDialogResult,
+} from '../../../../management/settings/api-portal-header/api-portal-header-edit-dialog/api-portal-header-edit-dialog.component';
+import { GioPermissionModule } from '../../../../shared/components/gio-permission/gio-permission.module';
+
+interface ApiForm {
+  apiKeyHeader: FormControl<string>;
+}
+
+@Component({
+  selector: 'portal-api-list',
+  standalone: true,
+  imports: [
+    GioPermissionModule,
+    MatButton,
+    MatCard,
+    MatCardContent,
+    MatCardHeader,
+    MatCardSubtitle,
+    MatCardTitle,
+    MatCell,
+    MatCellDef,
+    MatColumnDef,
+    MatHeaderCell,
+    MatHeaderRow,
+    MatHeaderRowDef,
+    MatIcon,
+    MatIconButton,
+    MatRow,
+    MatRowDef,
+    MatTable,
+    MatTooltip,
+    NgIf,
+    MatTableModule,
+    CdkDropList,
+    CdkDrag,
+    RouterLink,
+    FormsModule,
+    MatDialogContent,
+    MatFormField,
+    MatInput,
+    MatLabel,
+    ReactiveFormsModule,
+    MatFormFieldModule,
+    GioSaveBarModule,
+  ],
+  templateUrl: './portal-api-list.component.html',
+  styleUrl: './portal-api-list.component.scss',
+})
+export class PortalApiListComponent implements OnInit, OnDestroy {
+  @ViewChild('table', { static: true }) table: MatTable<ApiPortalHeader>;
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  private destroyRef = inject(DestroyRef);
+
+  settings: PortalSettings = {};
+  dataSource: ApiPortalHeader[] = [];
+  form: FormGroup<ApiForm>;
+  formInitialValues: unknown;
+  displayedColumns: string[] = ['order', 'name', 'value', 'actions'];
+
+  constructor(
+    public environmentApiHeadersService: EnvironmentApiHeadersService,
+    private matDialog: MatDialog,
+    private snackBarService: SnackBarService,
+    private portalSettingsService: PortalSettingsService,
+    private readonly permissionService: GioPermissionService,
+  ) {}
+
+  ngOnInit(): void {
+    this.initListSubscription();
+    this.getList();
+    this.initSettings();
+    this.portalSettingsService
+      .get()
+      .pipe(
+        tap((portalSettings) => {
+          this.settings = portalSettings;
+        }),
+        takeUntilDestroyed(this.destroyRef),
```

Comment:
Replace all `takeUntil(this.unsubscribe$)` with take until destroyed

Reply:
Done